### PR TITLE
Implementation of Cork

### PR DESCRIPTION
**Pull request by Karla-Isabel-Sandoval**
:warning: DO NOT MERGE! :warning:

### CODE REVIEW

File: lib/cork/board.rb [hunk 1]
```diff
@@ -89,9 +140,196 @@ def labeled(label, value, justification = 12)
...
+    # Prints a verbose message taking an optional verbose prefix and
+    # a relatvie indentation valid for the UI action in the passed block.
+    #
+    def notice(message)
+      line = "\n[!] #{message}"
+      line = line.green if ansi?
+      puts(line)
+    end
+
+    # @todo Clean interface.
```
Comment (orta) [inline lib/cork/board.rb]: any plans for this?

File: lib/cork/board.rb [hunk 2]
```diff
@@ -89,9 +140,196 @@ def labeled(label, value, justification = 12)
...
     private
 
     # @!group Helpers
+    #----------------------------------------------------------------------#
+
+    # @return [String] Wraps a string taking into account the width of the
+    # terminal and an option indent. Adapted from
+    # rubocop:disable Metrics/LineLength
+    # http://blog.macromates.com/2006/Wrapping-text wrapping-text-with-regular-expressions/
+    # rubocop:enable Metrics/LineLength
```
Comment (orta) [inline lib/cork/board.rb]: oh rubocop

File: lib/cork/board.rb [hunk 3]
```diff
@@ -89,9 +140,196 @@ def labeled(label, value, justification = 12)
...
+    #
+    # @param [String] txt    The string to wrap
+    #
+    # @param [String] indent The string to use to indent the result.
+    #
+    # @return [String]       The formatted string.
+    #
```
Comment (orta) [inline lib/cork/board.rb]: wait, this comment refers to nothing

File: lib/cork/board.rb [hunk 4]
```diff
@@ -89,9 +140,196 @@ def labeled(label, value, justification = 12)
...
+    # @param [String] verbose_prefix
+    #        See #message
+    #
+    # @param [FixNum] relative_indentation
+    #        The indentation level relative to the current,
+    #        when the message is printed.
+    #
+
+    # @!group Helpers
```
Comment (orta) [inline lib/cork/board.rb]: this one too?

File: lib/cork/board.rb (hunk 2 above)
Comment (segiddins) [inline lib/cork/board.rb]: this is disabled in the master CocoaPods configuration already

File: lib/cork/text_wrapper.rb [hunk 5]
```diff
@@ -41,17 +41,11 @@ def wrap_formatted_text(string, indent = 0, max_width = 80)
     # @param  [Fixnum] indent
     #         The number of spaces to insert before the string.
     #
-    # @param  [Fixnum] max_width
-    #         The maximum width to use to format the string if the terminal
+    # @param  [Fixnum] width
+    #         The width to use to format the string if the terminal
```
Comment (orta) [inline lib/cork/text_wrapper.rb]: `The width to use to format the string if the terminal` -> `The width to use when formatting the string in the terminal`

**Comment by orta** [comment]
Everything looks good in general, a lot go my questions are based on documentation things rather than the ruby specifics, seems all good :+1:

File: .rubocop.yml [hunk 6]
```diff
@@ -7,3 +7,6 @@ AllCops:
 
 ParameterLists:
   CountKeywordArgs: false
+
+Metrics/ModuleLength:
```
Comment (segiddins) [inline .rubocop.yml:11]: this should be covered by inheriting from the up-to-date CocoaPods config

File: lib/cork/board.rb [hunk 7]
```diff
@@ -1,14 +1,44 @@
+require 'colored'
+
 module Cork
+  # Provides support for UI output. Cork provides support for nested
+  # sections of information and for a verbose mode.
+  #
   class Board
     attr_reader :input
```
Comment (segiddins) [inline lib/cork/board.rb]: all these attrs need to be documented

File: spec/spec_helper.rb [hunk 8]
```diff
@@ -1,3 +1,10 @@
+require 'simplecov'
```
Comment (segiddins) [inline spec/spec_helper.rb:2]: `require 'bundler/setup'` should be before this

**Comment by Karla-Isabel-Sandoval** [comment]
Thanks @orta and @segiddins for reviewing and for the feedback!

File: lib/cork/text_wrapper.rb (hunk 5 above)
Comment (Karla-Isabel-Sandoval) [inline lib/cork/text_wrapper.rb]: @orta - In this comment do you mean add the message after the arrow?

File: .rubocop.yml (hunk 6 above)
Comment (Karla-Isabel-Sandoval) [inline .rubocop.yml]: Since coverage is provided by the up-to-date CocoaPods config should we squash lines 10-11?

File: lib/cork/board.rb (hunk 7 above)
Comment (Karla-Isabel-Sandoval) [inline lib/cork/board.rb]: Could you please provide one example to follow? Thanks @segiddins

**Comment by segiddins** [comment]
@Karla-Isabel-Sandoval please let us know when this PR is ready for final review

File: lib/cork/board.rb (hunk 1 above)
Comment (Karla-Isabel-Sandoval) [inline lib/cork/board.rb]: @segiddins Any plans for the above. Should we keep it?

Comment (segiddins) [inline lib/cork/board.rb]: I think you were the one who wrote the comment? So I have no idea

File: lib/cork/text_wrapper.rb (hunk 5 above)
Comment (Karla-Isabel-Sandoval) [inline lib/cork/text_wrapper.rb]: Yes. You did. Just fixed it.

File: lib/cork/board.rb (hunk 7 above)
Comment (orta) [inline lib/cork/board.rb]: Here's one : https://github.com/CocoaPods/Core/blob/master/lib/cocoapods-core/specification.rb#L43-L46

File: .rubocop.yml (hunk 6 above)
Comment (Karla-Isabel-Sandoval) [inline .rubocop.yml]: What are the steps to inherit from the up-to-date CocoaPods config as mentioned by segiddins?

Comment (segiddins) [inline .rubocop.yml:11]: There's a rake task in Rainforest for that

File: lib/cork/board.rb (hunk 1 above)
Comment (Karla-Isabel-Sandoval) [inline lib/cork/board.rb]: @segiddins - I took it from here https://github.com/CocoaPods/CocoaPods/blob/master/lib/cocoapods/user_interface.rb
https://www.dropbox.com/s/rgt5xs70jlgmj5m/Screenshot%202016-01-06%2011.18.52.png?dl=0

File: spec/spec_helper.rb (hunk 8 above)
Comment (Karla-Isabel-Sandoval) [inline spec/spec_helper.rb:2]: Added your comment but diff is not updating after I pushed it here.

File: .rubocop.yml (hunk 6 above)
Comment (Karla-Isabel-Sandoval) [inline .rubocop.yml:11]: Updated CocoaPods config to 1.0 and got rid of the old version.

Comment (Karla-Isabel-Sandoval) [inline .rubocop.yml:11]: @segiddins What is the rake task in Rainforest, that you are referring for?

Comment (orta) [inline .rubocop.yml:11]: Looks like this one, https://github.com/CocoaPods/Rainforest/blob/master/Rakefile#L467-L497

**Comment by orta** [comment]
Just spotted this is green - @Karla-Isabel-Sandoval do you want people to review this PR? ( If so - you should let us know )

File: .rubocop.yml (hunk 6 above)
Comment (segiddins) [inline .rubocop.yml:11]: since you've updated the cocoa pods config, please back out this part of the change

File: lib/cork/board.rb [hunk 9]
```diff
@@ -1,14 +1,59 @@
...
+
 module Cork
+  # provides support for UI output. Cork provides support for nested
+  # sections of information and for a verbose mode.
+  #
+
+  # @return [Podfile] The Podfile specification that contains the
+  # information of the Pods that should be installed.
+  #
+  attr_reader :podfile
```
Comment (segiddins) [inline lib/cork/board.rb]: this is CocoaPods-specific and doesn't belong in Cork

**Comment by Karla-Isabel-Sandoval** [comment]
Hi Orta,

Yes, a review would be great. Felix and I worked on it together for a bit..
but he is not to familiar with CocoaPods.. so it might be best to ask
someone else for help as well.


Karla Sandoval
@kisabelyogi
Skype: Karla.Isabel.Sandoval
831 334  3420

File: lib/cork/board.rb (hunk 9 above)
Comment (Karla-Isabel-Sandoval) [inline lib/cork/board.rb]: Lines 8-11 need to be cut out ?

Comment (orta) [inline lib/cork/board.rb]: yes, Cork is meant to not be just for CocoaPods

**Comment by Karla-Isabel-Sandoval** [comment]
@orta CodeCov and TravisCI seem to dislike the fact that I got rid of the pod file documentation. Major erring is happening now.

**Comment by orta** [comment]
You changed a lot of things at once in  df4d4f2, I don't really understand the changes - can you explain them?

**Comment by orta** [comment]
OK, so when I clone this locally I get the same build failures

**Comment by Karla-Isabel-Sandoval** [comment]
@orta, how do we fix it? what would you like to happen and what are the steps to make it happen?

**Comment by orta** [comment]
Well  df4d4f2 introduces a different public API for Cork - for example `bord.verbose?` used to be something you could do, and now you can't. As the specs weren't touched in that commit then the two are entirely out of sync. Which is why it's failing.

Which means, either fixing the specs with the new API or going back to the old API and just removing the `def podfile` that was brought up for review are the two solutions.

**Comment by orta** [comment]
So a version of just removing the Podfile, looks like this; https://github.com/CocoaPods/Cork/commit/a56fb4486a7258aee59554670f0dc1a11120e784 and is green.

I want to ship this PR, so maybe do that?

**Comment by Karla-Isabel-Sandoval** [comment]
@orta, sounds great. If removing the `def podfile`, makes it green, let's go with that.

**Comment by orta** [comment]
Deprecated by #7

**Pull request closed without merging**